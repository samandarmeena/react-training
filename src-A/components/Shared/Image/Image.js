import React from 'react';
import Proptypes from 'prop-types';

/**
 * Image component
 * It renders an image
 * @param {string} src
 * @param {string} alt
 * @return element
 */
const Image = (props) => {
  const { src, alt } = props;
  return <img src={src} alt={alt} />;
};

// proptype validations for props
Image.propTypes = {
  src: Proptypes.string.isRequired,
  alt: Proptypes.string.isRequired
};

export default Image;
