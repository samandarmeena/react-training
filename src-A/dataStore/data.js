import { IMAGE_URL } from '../constants/constants';
// Footer links array for FooterLinks component
const footerLinkStack = [
  [
    { headerTitle: 'Customer Service' },
    { linkTitle: 'Help Center' },
    { linkTitle: 'Contact Us' },
    { linkTitle: 'Return an Item' },
    { linkTitle: 'Recall Information' },
    { linkTitle: 'Safety Data Sheets' }
  ],
  [
    { headerTitle: 'Corporate Info' },
    { linkTitle: 'Staples Business Advantage Overview' },
    { linkTitle: 'Industries Served' },
    { linkTitle: 'Blog' },
    { linkTitle: 'Our Brands' }
  ],
  [
    { headerTitle: 'Staples Corporate Solutions' },
    { linkTitle: 'Office Products' },
    { linkTitle: 'Technology Products' },
    { linkTitle: 'Furniture' },
    { linkTitle: 'Facilities' },
    { linkTitle: 'Promotional Products' }
  ],
  [
    { headerTitle: 'New Customer' },
    { linkTitle: 'Become a Customer' },
    { linkTitle: 'Govt. Customers' },
    { headerTitle: 'Memberships' },
    { linkTitle: 'Join the Staples Team' },
    { linkTitle: 'Careers' }
  ]
];

// accessibility image for FooterConnect component
const accessibilityImg = {
  src: `${IMAGE_URL}essential-accessibility.png`,
  alt: 'Essential Accessibility'
};

// social connect image for FooterConnect component
const socialImgs = [
  { src: `${IMAGE_URL}linkedin.png`, alt: 'LinkedIn' },
  { src: `${IMAGE_URL}twitter.png`, alt: 'Twitter' },
  { src: `${IMAGE_URL}facebook.png`, alt: 'Facebook' },
  { src: `${IMAGE_URL}youtube.png`, alt: 'Youtube' },
  { src: `${IMAGE_URL}insta.png`, alt: 'Instagram' }
];

// copyright text for footer copyright
const copyrightText =
  'Copyright &copy; 2001-2019 Corporate Express Canada, Inc., a Staples company. All rights reserved.';

// copyright links for footer copyright
const copyRightLinks = [
  'Terms & Conditions',
  ' | ',
  'Privacy Policy',
  ' | ',
  'Staples Bussiness Advantage'
];

// export all data
export {
  footerLinkStack,
  accessibilityImg,
  socialImgs,
  copyRightLinks,
  copyrightText
};
