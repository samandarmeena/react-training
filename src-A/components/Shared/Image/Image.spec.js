import React from 'react';
import { mount } from 'enzyme';
import Image from './Image';
import { accessibilityImg } from '../../../dataStore/data';

describe('Image component', () => {
  let wrapper = '';

  beforeAll(() => {
    wrapper = mount(
      <Image src={accessibilityImg.src} alt={accessibilityImg.alt} />
    );
  });

  it('Image component rendered without an error', () => {
    expect(wrapper).toBeTruthy();
  });

  it('Check length of the children', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('Snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
