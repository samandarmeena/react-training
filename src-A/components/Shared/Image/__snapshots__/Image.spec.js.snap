// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Snapshot testing 1`] = `
<Image
  alt="Essential Accessibility"
  src="assets/images/essential-accessibility.png"
>
  <img
    alt="Essential Accessibility"
    src="assets/images/essential-accessibility.png"
  />
</Image>
`;
